packages/cadl-ranch-specs/http/azure/client-generator-core/access-namespace/main.tsp

```diff
@@ -0,0 +1,50 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using global.Azure.ClientGenerator.Core;
+
+@doc("Test for access/usage decorator.")
+@supportedBy("dpg")
+@scenarioService("/azure/client-generator-core/access-namespace")
+namespace _Specs_.Azure.ClientGenerator.Core.AccessNamespace;
+
+// this model should be access=internal and usage=output
+model OutputModel {
+  name: string;
+}
+
+// all the operations and models within this interface should be access=internal
+@scenario
+@scenarioDoc("""
+  This scenario contains access/usage on namespaces.
+  `internalOutput` operation should be internal. `OutputModel` model should be internal + output.
+  `Model1` and `Model2` models should be public + output.
+  
+  Expected response body:
+  ```json
+  {
+    "name": <any string>
+  }
+  ```
+  """)
+@access(Access.internal)
+namespace InternalOperation {
```

Comment:
should this test be added to `client-generator-core/access` as a new separate namespace?

Reply:
Do you mean put it into `client-generator-core/access/namespace/main.tsp` ? Or reverse it as `client-generator-core/namespace/access/main.tsp` ?